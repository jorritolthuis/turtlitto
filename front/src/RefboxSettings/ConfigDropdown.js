import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import Dropdown from "../Dropdown";

const MODE_VALUES = ["Practice", "Dual team", "Match"];
const CONFIG_VALUES = [
  "Role assigner on",
  "Role assigner off",
  "Pass demo",
  "Penalty mode",
  "Ball Handling demo"
];

/**
 * Constructs two dropdowns for the type of game and role
 * Author: G.W. van der Heijden
 * Author: S.A. Tanja
 * Author: T.T.P. Franken
 *
 * Props:
 * - value: current value of the dropdown
 * - onChange: a function on what to do when the button is pressed
 */
const ConfigDropdown = props => {
  return (
    <DropDownSection>
      <RefboxDropdown
        currentValue={props.value}
        values={MODE_VALUES}
        onChange={value => {
          props.onChange("mode", value);
        }}
        enabled={true}
      />
      <RefboxDropdown
        currentValue={props.value}
        values={CONFIG_VALUES}
        onChange={value => {
          props.onChange("role", value);
        }}
        enabled={true}
      />
    </DropDownSection>
  );
};

const DropDownSection = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  justify-content: center;
`;

const RefboxDropdown = styled(Dropdown)`
  width: 75%;
  margin-left: auto;
  margin-right: auto;
  height: 4rem;
`;

ConfigDropdown.propType = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default ConfigDropdown;
