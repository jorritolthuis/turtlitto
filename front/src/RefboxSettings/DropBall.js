import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

/**
 * Constructs a button for the Drop Ball command
 * Author: G.W. van der Heijden
 * Author: T.T.P. Franken
 *
 * Props:
 * - onClick: a function on what to do when the button is pressed
 */
const DropBall = props => {
  return (
    <DropBallWrapper className={props.className}>
      <DropBallButton
        id={"drop-ball-button"}
        onClick={() => {
          props.onClick();
        }}
      >
        DB
      </DropBallButton>
    </DropBallWrapper>
  );
};

const DropBallWrapper = styled.div`
  height: 25%;
  display: flex;
  flex-direction: row;
  justify-content: center;
`;

const DropBallButton = styled.button`
  background-color: ${props => props.theme.button};
  height: 75%;
  width: 25%;
  border: 0.25rem solid;
  font-size: 2rem;
  margin: auto;
  &:active {
    background-color: ${props => props.theme.buttonActive};
  }
`;

DropBall.propType = {
  onClick: PropTypes.func.isRequired
};

export default DropBall;
