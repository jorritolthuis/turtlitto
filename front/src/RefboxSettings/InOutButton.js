import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

/**
 * Constructs two buttons for in and out
 * Author: G.W. van der Heijden
 * Author: T.T.P. Franken
 *
 * Props:
 * - onClick: a function on what to do when the button is pressed
 */
const InOutButton = props => {
  return (
    <Container>
      <GoButton
        onClick={() => {
          props.onClick("in");
        }}
      >
        {"Go in"}
      </GoButton>
      <GoButton
        onClick={() => {
          props.onClick("out");
        }}
      >
        {"Go out"}
      </GoButton>
    </Container>
  );
};

const Container = styled.div`
  width: 50%;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const GoButton = styled.button`
  background-color: ${props => props.theme.inOutButton};
  height: 4rem;
  width: 75%;
  margin-left: auto;
  margin-right: auto;
  border: 0.25rem solid;
  font-size: 2rem;
  display: block;
  &:active {
    background-color: ${props => props.theme.inOutButtonActive};
  }
`;

InOutButton.propType = {
  onClick: PropTypes.func.isRequired
};

export default InOutButton;
