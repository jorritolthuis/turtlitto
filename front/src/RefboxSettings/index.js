import React from "react";
import DropBall from "./DropBall";
import InOutButton from "./InOutButton";
import styled from "styled-components";
import sendToServer from "../sendToServer";

/**
 * Gives the settings part of the refbox:
 * - Drop ball
 * - Go in
 * - Go out
 *
 * Author: G.W. van der Heijden
 * Author: S.A. Tanja
 * Author: T.T.P. Franken
 */
const RefboxSettings = props => {
  return (
    <Refbox className={props.className}>
      <DropBallButton
        onClick={() => {
          sendToServer("dropped_ball", "command", props.session);
        }}
      />
      <ButtonBlockWrapper>
        <InOutButton
          onClick={prop => {
            sendToServer(prop, "command", props.session);
          }}
        />
      </ButtonBlockWrapper>
    </Refbox>
  );
};

const ButtonBlockWrapper = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
  justify-content: center;
`;

const DropBallButton = styled(DropBall)`
  margin: 0.5rem;
`;

const Refbox = styled.div`
  height: 50%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0.5rem;
`;

export default RefboxSettings;
