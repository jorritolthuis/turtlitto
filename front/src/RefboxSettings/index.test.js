import React from "react";
import RefboxSettings from ".";
import { shallow } from "enzyme";

describe("RefboxSettings", () => {
  it("shows the settings including two dropdown menus and three buttons", () => {
    const wrapper = shallow(<RefboxSettings />);
    expect(wrapper).toMatchSnapshot();
  });
});
