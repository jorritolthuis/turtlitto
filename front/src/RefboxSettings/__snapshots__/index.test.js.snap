// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefboxSettings shows the settings including two dropdown menus and three buttons 1`] = `
<styled.div>
  <DropBall
    onClick={[Function]}
  />
  <styled.div>
    <InOutButton
      onClick={[Function]}
    />
    <ConfigDropdown
      onChange={[Function]}
      value="Whatever"
    />
  </styled.div>
</styled.div>
`;
